teps/0048-task-results-without-results.md

```diff
@@ -283,6 +284,49 @@ spec:
       value: $(tasks.propagate-image-name.results.image)
 ```
 
+#### Users using the Task should feel confident
+
+In the current scenario, there can be a case where a `Task` which claims to produce the result(s) but doesn't produces and the execution of `Task` is successful. This can confuse the user of the `Task`. If the user uses a `Task` then he/she should feel confident that the following `Task` claiming to produce the result is going to produce else it should fail. For example, the following Task fails as the result neither has any default value nor the step produces results.
```

Comment:
```suggestion
Today, a `Task` can declare a `Result` but not produce the `Result` and its execution will still be successful. If a subsequent `Task` attempts to use the `Result`, then it would fail (causing the `Pipeline` to fail). As such, a `Result` is optional when unused but required when used - this can be confusing to users when failures happen later in the `Pipeline`. Supporting default values for the optional `Results` would help prevent unexpected failures, and improve our `Task` interface`. 

For example, the following `Task` *add-task* declares a `Result` named *sum* but it does not produce it. An execution of `Task` *add-task* would succeed, however any subsequent `Tasks` attempting to use the `Result` *sum* would fail. Providing a default *sum* would help us avoid these unexpected failures, when there are reasonable default values.
```